controller/generic_controller.go

```diff
@@ -155,11 +153,31 @@ func (g *genericController) Sync(ctx context.Context) error {
 	return g.sync(ctx)
 }
 
-func (g *genericController) sync(ctx context.Context) error {
+func (g *genericController) sync(ctx context.Context) (retErr error) {
 	if g.synced {
 		return nil
 	}
 
+	if g.queue == nil {
```

Comment:
Is this protected elsewhere higher up the chain such that we dont need to put a lock around this reading and then setting of g.queue?

Reply:
resolved offline. there is a lock further up the chain